README.md

```diff
@@ -48,3 +39,118 @@ The code layout consists mainly of:
    - `transform_*.py` - transform the extracted raw data into output formats
 - `tests/` - the unit tests
 
+## File formats
+This scraper generates three different JSON file types.
+
+### `revisions.json`
+This file contains the revision hashes of the changesets the probe files were scraped. These hashes are mapped to an human-readable version string.
+
+```
+{
+  "<channel>": {
+    "<revision hash>": {
+      "version": "<human-readable version string>"
+    },
+    ...
+  },
+  ...
+  "aurora": {
+    "1196bf3032e1bce1fb07a01fd9082a767426c5fb": {
+      "version": "51"
+    },
+  },
+  ...
+}
+```
+
+### `general.json`
+This file contains general properties related to the scraping process. As of today, it only contains the `lastUpdate` property, which is the day and time the scraping was performed, in ISO 8601 format.
+
+```
+{
+  "lastUpdate": "2018-01-15T17:57:08.944690+01:00"
+}
+```
+
+### Probe data file
+This file contains the data for the probes. The data might be spread across multiple files. It has the following format:
+
+```
+{
+  "<probe type>/<probe name>": {
+    "history": {
+      "<channel>": [
+        {
+          "cpp_guard": <string or null>,
+          "description": "<string>",
+          "details": {
+            "<type specific detail>": "<detail data>",
+            ...
+            "record_in_processes": [
+              "<string>",
+              ...
+            ]
+          },
+          "expiry_version": "<string>",
+          "optout": <bool>,
+          "revisions": {
+            "first": "<string>",
+            "last": "<string>"
+          },
+          "versions": {
+            "first": "<string>",
+            "last": "<string>"
+          }
+        },
+        ...
+      ]
+    },
+    "name": "<probe name>",
+    "type": "<probe type>"
+  },
+  ...
+  "histogram/A11Y_CONSUMERS": {
+    "history": {
+      "nightly": [
+        {
+          "cpp_guard": null,
+          "description": "A list of known accessibility clients that inject into Firefox process space (see https://dxr.mozilla.org/mozilla-central/source/accessible/windows/msaa/Compatibility.h).",
+          "details": {
+            "high": 11,
+            "keyed": false,
+            "kind": "enumerated",
+            "low": 1,
+            "n_buckets": 12
+          },
+          "expiry_version": "never",
+          "optout": true,
+          "revisions": {
+            "first": "320642944e42a889db13c6c55b404e32319d4de6",
+            "last": "6f5fac320fcb6625603fa8a744ffa8523f8b3d71"
```

Comment:
Now that I see an example entry, I'm a bit concerned about ease of getting "probes active in the current latest release in each channel", which is a pretty common pattern

Reply:
The [probe dictionary](https://georgf.github.io/fx-data-explorer/index.html?search=&searchtype=in_any&optout=true&channel=release&constraint=is_in&version=57) does this by getting all the probes for all the channels and then filtering for `revision.first == <current revision>`. I think we can probably make it better though: do you want to file another issue on this repo?